src/gc/unix/gcenv.unix.cpp

```diff
@@ -449,7 +449,8 @@ the processors enabled.
 --*/
 static uintptr_t GetFullAffinityMask(int cpuCount)
 {
-    return ((uintptr_t)1 << (cpuCount)) - 1;
+    assert((cpuCount > 0) && (cpuCount <= sizeof(uintptr_t) * 8));
+    return (((uintptr_t)1 << (cpuCount - 1)) << 1) - 1;
```

Comment:
It is not obvious why the shift is split into two parts. It can use a comment.

Also, it may be better for readability to just use `if (cpuCount >= sizeof(uintptr_t) * 8) ...`.